src/test/java/htsjdk/samtools/SAMFileWriterFactoryTest.java

```diff
@@ -304,4 +322,48 @@ public void testAsync() throws IOException {
         writer = builder.setUseAsyncIo(false).makeWriter(header, false, outputFile, referenceFile);
         Assert.assertFalse(writer instanceof AsyncSAMFileWriter, "testAsync option=unset");
     }
+
+    @Test
+    public void testMakeSamOrBam_withSamExtension() throws IOException {
+        File tmpFile = File.createTempFile("testMakeSamOrBam_withCramExtension_", ".sam");
```

Comment:
No need for trailing underscore in the name.  Also, the variables here and elsewhere can be `final`